aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -183,6 +185,86 @@ static TensorIterator make_reduction(
   return TensorIterator::reduce_op(viewed_result1, viewed_result2, self.to(dtype));
 }
 
+template<typename scalar_t, typename func_t>
+static void _cumop_out_cpu_template(Tensor& result, const Tensor& self,
+    int64_t dim, scalar_t init_v, func_t && op) {
+  auto wrap_dim = maybe_wrap_dim(dim, self.dim());
+  if (result.sizes() != self.sizes()) {
+    result.resize_as_(self);
+  }
+  if (self.numel() == 0) {
+    return;
+  }
+  const auto input_shape = self.sizes();
+  const auto input_ndim = self.dim();
+
+  if (input_ndim == 0) {
+    result.fill_(self);
+    return;
+  }
+
+  auto temp_tensor = result;
+  if (!result.is_contiguous()) {
+    temp_tensor = temp_tensor.contiguous();
+  }
+
+  auto n = input_shape[wrap_dim];
+  const int64_t M = std::accumulate(
+      input_shape.cbegin(),
+      input_shape.cbegin() + wrap_dim,
+      1LL,
+      std::multiplies<int64_t>()
+      );
+  const int64_t N = std::accumulate(
+      input_shape.cbegin() + wrap_dim + 1,
+      input_shape.cend(),
+      1LL,
+      std::multiplies<int64_t>()
+      );
+
+  auto self_ptr = self.data_ptr<scalar_t>();
+  auto out_ptr = temp_tensor.data_ptr<scalar_t>();
+  auto offset = N * n;
+  parallel_for(0, M, internal::GRAIN_SIZE, [&](int64_t begin, int64_t end) {
+    auto s_index =  offset * begin;
+    for (int64_t f = begin; f < end; ++f) {
+      for (auto j = 0; j < N; j++ ) {
+        scalar_t cum_number = init_v;
```

Comment:
You might get precision problem here, as original code had accumulator of `accreal` type

Reply:
Discussed with @ngimel, we are fine using scalar_t here for acc.